torch/quantization/fx/prepare.py

```diff
@@ -273,8 +275,12 @@ def maybe_insert_input_observer_for_arg_or_kwarg(
         # regular flow for most nodes, except standalone modules
         is_weight = node_arg_is_weight(node, arg)
         assert qconfig is not None
+
         act_post_process_ctr = qconfig.weight if is_weight else \
             qconfig.activation
+        if isinstance(qconfig, EqualizationQConfig) and not is_weight:
```

Comment:
I'd recommend rewriting this so the `act_post_process_ctr` value gets set exactly once and is not modified, to minimize the chance of future bugs.  For example,

```
if is_weight:
  ...
elif isinstance(qconfig, EqualizationQConfig):
  ...
else:
  ...
```